src/protocol/coap.rs

```diff
@@ -0,0 +1,191 @@
+//! CoAP Binding Template
+
+use crate::extend::ExtendableThing;
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+use serde_with::{serde_as, skip_serializing_none};
+
+/// CoAP request method
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum Method {
+    Get,
+    Put,
+    Post,
+    Delete,
+    Patch,
+    Fetch,
+    #[serde(rename = "iPATCH")]
+    Ipatch,
+}
+
+/// CoAP Allowed block size
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize_repr, Serialize_repr)]
```

Comment:
```suggestion
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Deserialize_repr, Serialize_repr)]
```

I'm not sure if it should derive `Ord` and `PartialOrd` or not.